lib/storage/metadata/oplog/BucketdOplogInterface.js

```diff
@@ -0,0 +1,257 @@
+/*
+ * Main interface for bucketd oplog management
+ */
+const async = require('async');
+const { RESTClient: BucketClient } = require('bucketclient');
+const { jsutil, errors } = require('arsenal');
+const LogConsumer = require('arsenal/lib/storage/metadata/bucketclient/LogConsumer');
+const { isMasterKey } = require('arsenal/lib/versioning/Version');
+const OplogInterface = require('./OplogInterface');
+
+class BucketdOplogInterface extends OplogInterface {
+
+    constructor(params) {
+        super(params);
+        this.stopAt = -1;
+        this.backendRetryTimes = 3;
+        this.backendRetryInterval = 300;
+        this.bucketdOplogQuerySize = 20;
+        let bkBootstrap = ['localhost:9000'];
+        if (params && params.bootstrap !== undefined) {
+            bkBootstrap = params.bootstrap;
+        }
+        if (params && params.stopAt !== undefined) {
+            this.stopAt = params.stopAt;
+        }
```

Comment:
new JS features: optional chaining + nullish coalescing

```suggestion
        this.backendRetryTimes = 3;
        this.backendRetryInterval = 300;
        this.bucketdOplogQuerySize = 20;
        this.stopAt = params?.stopAt ?? -1;
        const bkBootstrap = params?.bootstrap ?? ['localhost:9000'];
```